python_modules/libraries/dagster-aws/dagster_aws/s3/io_manager.py

```diff
@@ -100,11 +103,68 @@ def handle_output(self, context, obj):
         context.add_output_metadata({"uri": MetadataValue.path(path)})
 
 
+class S3IOManagerResource(IOManagerFactoryResource[PickledObjectS3IOManager]):
```

Comment:
even thought it is more verbose, I do think that it is more consistent and clear to call this `PickledObjectS3IOManagerResource`